src/nxdoc/nuxeo-server/administration/variables-and-configuration/variables-and-configuration-pattern.md

```diff
@@ -0,0 +1,300 @@
+Note that `TEMPLATE_DIRECTORY` can either stand for the template name (in which case the sub-directory with this name will be resolved inside the `NUXEO_HOME/templates/` directory), or an absolute directory path.
+
+These steps are covered by environment variables expansion explained below.
+
+### Pre-processing Environment Variables Expansion
+
+During preprocessing, properties pushed to the main map can be [expanded](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java#L698) when they are declared with the pattern `${env:VARIABLE_NAME}` or `${env:VARIABLE_NAME:defaultValue}`, for `nuxeo.conf` templates and for properties held by templates.
+
+Values are replaced according to environment variables: for the given property name, the value returned by System#getEnv will be used, and will default to the `defaultValue` if specified, or to an empty string.
+
+Although all resolutions can follow this pattern, this is specifically useful to declare additional Nuxeo configuration templates:
+
+```
+nuxeo.templates=default,common,testinclude,${env:TEST_ENV:testenv}
+```
+
+Note that in the `nuxeo.conf` file, only properties referring to templates are covered by this expansion: all properties might need to be covered too, to be consistent with template properties resolution.
+
+Template properties resolution was introduced to cover a deployment use case, so that a docker image can be run in a specific target environment, with a pre-constructed map of values that can/should be controlled by the environment, in the `docker run` command line:
+
+```
+docker run -e NUXEO_ENVIRONMENT=prod -e NUXEO_PROFILES=the-deploy-template-name image
+```
+
+This strategy is used for explorer deployment with a docker image on an infrastructure run by IT, exposing a template named `explorer-deploy`. Extract from the [nuxeo.prod](https://github.com/nuxeo/nuxeo-explorer/blob/master/packages/nuxeo-platform-explorer-package/src/main/resources/install/templates/explorer-deploy/nuxeo.prod) file:
+
+```
+## all templates to apply on target deployment env
+nuxeo.template.includes=default,docker,s3binaries,postgresql,explorer-sitemode,explorer-virtualadmin
+
+## explorer confs mapping
+
+org.nuxeo.apidoc.apidocAdmin.password=${env:ORG_NUXEO_APIDOC_APIDOCADMIN_PASSWORD:apidocAdmin}
+
+## default confs mapping
+
+# publicly visible URL
+nuxeo.url=${env:NUXEO_URL:http://localhost:8080/nuxeo}
+
+org.nuxeo.ecm.product.name=Nuxeo Explorer
+
+[...]
+```
+
+An alternative to this strategy would be to add to the map of properties, as a last step during pre-processing, all environment variables that would start with `nuxeo` (for instance). 
```

Comment:
Same, this was a reference to an existing jira issue and should not be documented since it's not done (and is not currently planned)